src/ServiceManagement/Management/Management.Tests/SubscriptionPrincipalTest.cs

```diff
@@ -0,0 +1,95 @@
+﻿//
+// Copyright (c) Microsoft.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+using Microsoft.Azure.Test.HttpRecorder;
+using Microsoft.WindowsAzure.Testing;
+
+namespace Microsoft.WindowsAzure.Management.Testing
+{
+    using System;
+    using System.Linq;
+    using Management;
+    using Management.Models;
+    using Microsoft.Azure.Test;
+    using Xunit;
+    public class SubscriptionPrincipalTest : TestBase
+    {
+        [Fact]
+        public void SubscriptionCRUD()
+        {
+            using (var undoContext = UndoContext.Current)
+            {
+                undoContext.Start();
+                var mgmt = this.GetManagementClient();
+
+                // Retrive existing SPN's if any
+                var existingSPNs = mgmt.SubscriptionServicePrincipals.List();
+
+                // Generate a Random SPN Object Id
+                string newSPNObjectId = Guid.NewGuid().ToString();
```

Comment:
>Guid.NewGuid().ToString(); [](start = 40, length = 26)

I'm not sure this will work in tests since it will generate a new guid each time. Can you use a guid property with constant value here?

Reply:
Yes it won't work in Playback mode. Changed to static value used in Record mode.